File: Submissions/121_Best_Time_to_Buy_and_Sell_Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if(prices.empty()) return 0;
        
        int min_price = prices.front();
        int result = 0;
        for(auto& price: prices){
            if(min_price > price) {
                min_price = price;
            } else {
                result = max(result, price - min_price);
            }
        }
        return result;
    }
};

File: Kotlin/121_Best_Time_to_Buy_and_Sell_Stock.kt
class Solution {
    fun maxProfit(prices: IntArray): Int {
        var result = 0
        var minPrice = Int.MAX_VALUE
        
        prices.forEach {
            minPrice = minOf(minPrice, it)
            result = maxOf(result, it - minPrice)
        }
        
        return result
    }
}